content/blog/2022/07/verify-signatures-with-gha-and-prefix.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Verify image signatures with GithubActions and KeylessPrefix 
+authors:
+- Raul Cabello Martin 
+date: 2022-07-14
+---
+
+With the latest Kubewarden release v1.1.0 and the latest release of
+the [verify-image-signatures](https://github.com/kubewarden/verify-image-signatures) policy, it's now possible to use
+`GithubActions` or `KeylessPrefix` for verifying images. Read our
+previous [blog post](https://www.kubewarden.io/blog/2022/05/verifying-image-signatures/) if you want to learn more about
+how to verify container images with Sigstore using Kubewarden.
+
+## Let's see it in action!
+
+We want to verify the image `ghcr.io/raulcabello/app-example` which was signed inside a GitHub action. Let's run the
+following command to see the issuer and subject:
+
+```
+COSIGN_EXPERIMENTAL=1 cosign verify ghcr.io/raulcabello/app-example:v0.4.0
+```
+
+The output is:
+
+```json 
+{
+  ....
+  "Issuer": "https://token.actions.githubusercontent.com",
+  "Subject": "https://github.com/raulcabello/app-example/.github/workflows/ci.yml@refs/tags/v0.4.0"
+}
+```
+
+For this example, a Kubernetes cluster with Kubewarden already installed is required. The installation process is
+described in the [quick start guide](https://docs.kubewarden.io/quick-start).
+
+Let's verify this image with the new signature types: GitHubAction and KeylessPrefix
+
+### GitHubAction
+
+```yaml 
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+name: verify-image-signatures-policy
+spec:
+module: registry://ghcr.io/kubewarden/policies/verify-image-signatures:v0.1.4
+rules:
+- apiGroups: [""]
+  apiVersions: ["v1"]
+  resources: ["pods"]
+  operations:
+    - CREATE
+    - UPDATE
+      mutating: true
+      settings:
+      signatures:
+        - image: "ghcr.io/raulcabello/app-example:*" # match all tags 
+          github_actions:
+            - owner: "raulcabello"
+              repo: "app-example" #optional. if omited all apps from the owner will ve valid.
+```
+
+This policy verifies all containers with an image that is `ghcr.io/raulcabello/app-example`. It will accept
+containers that were signed inside a GitHub action for the owner `raulcabello` and the app `app-example`.
+
+### KeylessPrefix
```

Comment:
I would not show how to do keyless prefix verification to check GitHub action, because this can be insecure. I'm referring to the shared workflows unwanted side effect that we discovered.

Talking about that, I think it would be even more relevant to have a section explaining why you should **not** use `KeylessPrefix`. It would be nice to add some example, like from our own policies that are actually signing policies in that way.